### PR TITLE
docs: update changelog for 0.11.0

### PR DESCRIPTION
**Pull request by henryiii**
Starting to get ready.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -11,15 +11,42 @@ Unreleased
   (PR :pr:`567`)
 - Added ``runner`` parameter to ``util.project_wheel_metadata``
   (PR :pr:`566`, Fixes :issue:`553`)
```
Comment (layday) [inline CHANGELOG.rst:13]: For consistency:

```suggestion
  (PR :pr:`566`, fixes issue :issue:`553`)
```

**Review by layday — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -122,7 +122,6 @@ filterwarnings = [
   "error",
   "ignore:path is deprecated.:DeprecationWarning",
   "ignore:The --rsyncdir command line argument and rsyncdirs config variable are deprecated.:DeprecationWarning",
-  "ignore:Python 3.14 will, by default, filter extracted tar archives:DeprecationWarning",
```
Comment (henryiii) [inline pyproject.toml:125]: I think flit is triggering this? Pretty sure it's not us. I might try only ignoring it for Flit, if that doesn't work, I'll just restore it.

Comment (henryiii) [reply pyproject.toml:125]: Ahh, it's actually in our tests. Okay, will put it back for now. We maybe can silence this in our tests in the future.

**Review by gaborbernat — APPROVED** [review]
We should make it 1.0.0 instead of 0.11.0.

**Pull request merged**